tests/test_marathon.py

```diff
@@ -0,0 +1,449 @@
+import re
+
+import jsonschema
+import requests
+
+from dcos import http, marathon
+from dcos.errors import DCOSException, DCOSHTTPException
+
+import mock
+import pytest
+
+
+def test_add_pod_puts_json_in_request_body():
+    _assert_add_pod_puts_json_in_request_body(pod_json={"some": "json"})
+    _assert_add_pod_puts_json_in_request_body(
+        pod_json=["another", {"json": "value"}])
+
+
+def test_add_pod_returns_parsed_response_body():
+    _assert_add_pod_returns_parsed_response_body({"id": "i-am-a-pod"})
+    _assert_add_pod_returns_parsed_response_body(["another", "pod", "json"])
```

Comment:
The above two tests are the only ones that directly exercise `marathon.Client.add_pod`. The remaining tests are for the infrastructure of `RpcClient`, and so apply to _all_ methods of `marathon.Client`.